students/annguan/lesson04/mailroom2.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+
+### Lesson 4 Mailroom, Part 2, uses dict
+
+from textwrap import dedent
+import os
+import sys
+import math
+
+# a data structure holds a list of donors and a history of the amounts they have donated.
+def get_donor_db():
+    return {'Abraham Lincoln': ("Abraham Lincoln", [145674.32, 23465]),
+'Barack Obama': ("Barack Obama", [3456324.11, 3495, 323]), 
+'Charlie Brown':("Charlie Brown",[453.67]),
+'Doctor Who': ("Doctor Who", [5600, 42]),
+'Eve WallE': ("Eve WallE",[2.22])}
+
+# Write thank you letters to disk as a text file
+def write_letters_disk():
+    for donor in donor_db.values():
+        letter = comp_letter(donor)
+        filename = donor[0].replace(" ","_")+".txt"
+        open(filename,'w').write(letter)
```

Comment:
prefer to use `with` when working with files to ensure that files get closed.

Reply:
Implemented the following:
        with open(filename, 'w') as file:
            file.write(letter)